### ISSUE TITLE
A Layer that Sets K Smallest Values to Zero

### ISSUE THREAD

**Issue by PkuClosed** [fchollet/keras#2867]
Hi,

I am trying to have a layer with a very simple function. I want to set the K smallest entries in a vector to zero. For example, when K = 2 and the input vector is (1,2,3,4,5), the output will be (0,0,3,4,5). Conceptually this is very simple, but I am wondering if there is an existing tool in Keras for this, or if there is an easy way to realize such a function. I looked at Masking and Pooling layers, but they don't seem to be relevant.

Thanks,
Chuyang

**Comment by xingdi-eric-yuan** [comment]
if you're using theano backend, then you can use T.argsort()

**Comment by speedcell4** [comment]
But how about Tensorflow backend?
